Services/CO.CDP.EntityVerification/Ppon/OrganisationRegisteredEventHandler.cs

```diff
@@ -1,21 +1,33 @@
 using CO.CDP.EntityVerification.Events;
 using CO.CDP.EntityVerification.Persistence;
 using CO.CDP.MQ;
-using Microsoft.AspNetCore.Components;
-using System.Transactions;
+using Identifier = CO.CDP.EntityVerification.Persistence.Identifier;
 
 namespace CO.CDP.EntityVerification.Ppon;
 
-public class OrganisationRegisteredEventHandler(IPponService pponService, IPponRepository pponRepository, IPublisher publisher)
+public class OrganisationRegisteredEventHandler(IPponService pponService,
+    IPponRepository pponRepository)
     : IEventHandler<OrganisationRegistered>
 {
     public async Task Handle(OrganisationRegistered @event)
     {
-        var pponId = pponService.GeneratePponId();
+        Persistence.Ppon newPpon = new() {
+            PponId = pponService.GeneratePponId(),
+            Name = @event.Name,
+            OrganisationId = @event.Id
+        };
 
-        Persistence.Ppon newIdentifier = new() { PponId = pponId };
+        List<Identifier> ids = [];
+        foreach (var e in @event.AllIdentifiers())
+        {
+            ids.Add(new Identifier { Ppon = newPpon,
+                LegalName = e.LegalName,
+                Scheme = e.Scheme,
+                Uri = e.Uri });
+        }
 
-        pponRepository.Save(newIdentifier);
-        await publisher.Publish(newIdentifier);
```

Comment:
I just realised, we need a proper event type to publish that is separate from the Ppon model. We must not be publishing EF entities!

Reply:
This is something for the other story.